akka-actor/src/main/scala/akka/actor/dungeon/ReceiveTimeout.scala

```diff
@@ -79,4 +80,15 @@ private[akka] trait ReceiveTimeout { this: ActorCell =>
       receiveTimeoutData = (receiveTimeoutData._1, emptyCancellable)
     }
 
+  private def messageMarkedToNotInfluenceTimeout(message: Any): Boolean = message match {
+    case _: NotInfluenceReceiveTimeout => true
+    case w: WrappedMessage =>
+      w.message match {
```

Comment:
maybe use `WrappedMessage.unwrap(message)` to recursively unwrap. Would still need the scala.util.Success case, unless you add that to `WrappedMessage.unwrap`.